### PR TITLE
fix(preset-mini): support group-data variant

### LINKED ISSUE #2827: group-data doesn't work
### UnoCSS version

0.53.4

### Describe the bug

I'm new to UnoCSS (first project using it) so pardon me if I'm doing something wrong.

I noticed that group-data selectors that work in Tailwind don't seem to work with UnoCSS:

```
<div class="group" data-state="open">
  <span class="group-data-[state=open]:text-red-400">
    hello
  </span>
</div>
```

(Text is supposed to turn red. Instead, the CSS output is simply empty.)

### Reproduction

- [Tailwind Playground](https://play.tailwindcss.com/JrEeGTGxrT) (works)
- [UnoCSS Playground](https://unocss.dev/play/?html=DwEwlgbgBAxgNgQwM5ILwCIDmAnA9gVwAd0oQEAXBAWiUvIFMNdD6A7dAPgCgopglCCVrEQoMOAoSplKVANq0KjZmwC6ALgYAPclWz0QVACwAGE5x68oAC3pw4uS8AD0AodxfgIHIA&config=JYWwDg9gTgLgBAbwFBzgEwKYDNgDsMDCEuOA5gDQpxhQYDOGMAgjDFMAEYCuMwWAnpVQ16jAJIBjYnSHVaDGAFVcESgF84WKBBBwA5FxUS6dPUiQYAHpFjpsAQy4AbeJhz4iJYKQAUyVFDO9ABccADaVKhhehJcdDA6ALSBThh65IhwUk7QoXq0aHpwagC6VCWydAAW0DCxMHSh-qj6sfFJ1bX1enkwVjCJTqRwfZYD0Pa4pBhwNQBuGFDBowN99k5mqGqyIgqN4ZFyokoqPgCUssLyjCxsnDx8-OeXRwqS0n6HqHQS6xihAEYAHQAJheqAkaFweSqrDAjQA9Aj6CAgdUEZsWmoLuUkNikEA&css=Q&options=N4XyA) (doesn't work)

### System Info

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/unocss/unocss/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't [already an issue](https://github.com/unocss/unocss/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/unocss/unocss/discussions) or join our [Discord Chat Server](https://chat.antfu.me).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by zouhangwithsweet**
fix #2827

### CODE REVIEW

**Comment by zouhangwithsweet** [comment]
@zyyv Hi, Chris. Can you help to check the test cases? I have not idea about how to fix them. :D

**Comment by zyyv** [comment]
> @zyyv Hi, Chris. Can you help to check the test cases? I have not idea about how to fix them. :D

ok, let me check it.

**Comment by zyyv** [comment]
@zouhangwithsweet hi, I haven't checked the cause of the error in the test yet. But I think this PR change is too targeted. We should enhance `taggedPseudoClassMatcher` to support more modifiers.
https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state

**Comment by zouhangwithsweet** [comment]
> @zouhangwithsweet hi, I haven't checked the cause of the error in the test yet. But I think this PR change is too targeted. We should enhance `taggedPseudoClassMatcher` to support more modifiers. https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state

Agreed! Maybe I need more time to understand source code.

File: packages/preset-mini/src/_variants/data.ts [hunk 1]
```diff
@@ -18,3 +18,20 @@ export const variantDataAttribute: VariantObject = {
...
+  name: 'group-data',
+  match(matcher, ctx: VariantContext<Theme>) {
+    const variant = variantGetParameter('group-data-', matcher, ctx.generator.config.separators)
+    if (variant) {
+      const [match, rest] = variant
+      const dataAttribute = h.bracket(match) ?? ctx.theme.data?.[match] ?? ''
+      if (dataAttribute) {
+        return {
+          matcher: rest,
+          selector: s => `.group[data-${dataAttribute}] ${s}`,
```
Comment (antfu) [inline packages/preset-mini/src/_variants/data.ts:32]: This should respect the `attributifyPseudo` option

**Comment by zouhangwithsweet** [comment]
I found that `group-[[data-state=open]]:text-red-400` is supported.
Just change `group-data-[state=open]:xx` to `group-[[data-state=open]]:xx`. :D

**Comment by antfu** [comment]
Test is failing, can you have a look?

**Pull request merged**